File: app/src/main/java/com/xfl/kakaotalkbot/AppData.java
package com.xfl.kakaotalkbot;

import org.mozilla.javascript.ScriptableObject;
import org.mozilla.javascript.annotations.JSStaticFunction;

public class AppData extends ScriptableObject {
    public AppData() {
    }

    @JSStaticFunction
    public static void putBoolean(String key, boolean bool) {
        MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).edit().putBoolean(key, bool).apply();
    }

    @JSStaticFunction
    public static boolean getBoolean(String key, boolean defaultValue) {
        return MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).getBoolean(key, defaultValue);
    }

    @JSStaticFunction
    public static void putInt(String key, int integer) {
        MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).edit().putInt(key, integer).apply();
    }

    @JSStaticFunction
    public static Integer getInt(String key, int defaultValue) {
        return MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).getInt(key, defaultValue);
    }

    @JSStaticFunction
    public static void putString(String key, String string) {
        MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).edit().putString(key, string).apply();
    }

    @JSStaticFunction
    public static String getString(String key, String defaultValue) {
        return MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).getString(key, defaultValue);
    }

    @JSStaticFunction
    public static void remove(String key) {
        MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).edit().remove(key).apply();
    }

    @JSStaticFunction
    public static void clear() {
        MainApplication.getContext().getSharedPreferences("sharedbotdata", 0).edit().clear().apply();
    }

    @Override
    public String getClassName() {
        return "AppData";
    }
}

File: app/src/main/java/com/xfl/kakaotalkbot/AppData.kt
package com.xfl.kakaotalkbot

import org.mozilla.javascript.ScriptableObject
import org.mozilla.javascript.annotations.JSStaticFunction

class AppData : ScriptableObject() {

    override fun getClassName(): String {
        return "AppData"
    }

    companion object {

        @JSStaticFunction
        fun putBoolean(key: String, bool: Boolean) {
            MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).edit().putBoolean(key, bool).apply()
        }

        @JSStaticFunction
        fun getBoolean(key: String, defaultValue: Boolean): Boolean {
            return MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).getBoolean(key, defaultValue)
        }

        @JSStaticFunction
        fun putInt(key: String, integer: Int) {
            MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).edit().putInt(key, integer).apply()
        }

        @JSStaticFunction
        fun getInt(key: String, defaultValue: Int): Int? {
            return MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).getInt(key, defaultValue)
        }

        @JSStaticFunction
        fun putString(key: String, string: String) {
            MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).edit().putString(key, string).apply()
        }

        @JSStaticFunction
        fun getString(key: String, defaultValue: String): String {
            return MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).getString(key, defaultValue)
        }

        @JSStaticFunction
        fun remove(key: String) {
            MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).edit().remove(key).apply()
        }

        @JSStaticFunction
        fun clear() {
            MainApplication.context!!.getSharedPreferences("sharedbotdata", 0).edit().clear().apply()
        }
    }
}
